### DISCUSSION TITLE
How can I encrypt and decrypt an ID

### DISCUSSION THREAD

**Discussion by ehary-devs** [filamentphp/filament#9435] [Help]
How can I encrypt and decrypt an ID in a URL like http://localhost:4444/admin/users/1/edit for enhanced security, ensuring that the ID is not directly exposed?

**Comment by dissto** [comment]
Use ulid or uuids as primary key instead of the default unsignedBigInteger

https://laravel.com/docs/10.x/migrations#column-method-ulid

https://laraveldaily.com/post/laravel-users-table-change-primary-key-id-to-uuid

**Reply by ehary-devs** [reply]
thank you, I've tried this before, I understand how to use it, only I use the shield plugin, there are lots of ID relationships to users. I'm confused about using it.

**Reply by dissto** [reply]
Well the downside is that you need to modify the migrations that reference the id because for example in the case of ulid it's no longer `$table->foreignId` but `$table->foreignUlid` and so on. 

That means you would need to change migrations from plugins too
